File: project_euler/Problem050/Problem050.java
package project_euler;

import java.util.ArrayList;
import java.util.List;

public class Problem050 {
    public void solution() {
        int target = 1000000;
        List<Integer> primes = new ArrayList<>();
        primes.add(2);
        int index = 3;
        int summation = 2;
        while (summation < target) {
            if (isPrime(index)) {
                primes.add(index);
                summation += index;
            }
            index += 2;
        }

        int length = primes.size();
        for (int i = length - 1; i > 0; --i) {
            int value = summation - primes.get(i);
            if (isPrime(value)) {
                System.out.println(value);
                return;
            }
            summation = value;

            for (int j = i; j < length; ++j) {
                value += primes.get(j) - primes.get(j - i);
                if (isPrime(value) && value < target) {
                    System.out.println(value);
                    return;
                }
            }
        }
    }

    private boolean isPrime(int value) {
        if (1 != value % 6 && 5 != value % 6) {
            return 2 == value || 3 == value;
        }

        int border = (int) Math.sqrt(value);
        for (int i = 5; i <= border; i += 6) {
            if (0 == value % i || 0 == value % (i + 2)) {
                return false;
            }
        }
        return 1 != value;
    }
}

File: project_euler/Problem050/Problem050.py
import math

class Problem050:
    def solution(self) -> None:
        target, primes, index, summation = 1000000, [2], 3, 2
        while summation < target:
            if self.__is_prime(index):
                primes.append(index)
                summation += index
            index += 2

        length = len(primes)
        for i in range(length - 1, -1, -1):
            value = summation - primes[i]
            if self.__is_prime(value):
                print(value)
                return
            summation = value

            for j in range(i, length):
                value += primes[j] - primes[j - i]
                if self.__is_prime(value) and value < target:
                    print(value)
                    return

    def __is_prime(self, value: int) -> bool:
        if 1 != value % 6 and 5 != value % 6:
            return 2 == value or 3 == value

        border = int(math.sqrt(value))
        for i in range(5, border + 1, 6):
            if not value % i or not value % (i + 2):
                return False
        return 1 != value
